src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -7196,5 +7202,974 @@ static void M(ReadOnlySpan<int> s)
+                //         return s1 && s2; // 2
+                Diagnostic(ErrorCode.ERR_EscapeCall, "s1 && s2").WithArguments("S.operator &(in S, in S)", "x").WithLocation(11, 16),
+                // (23,16): error CS8352: Cannot use variable 's' in this context because it may expose referenced variables outside of their declaration scope
+                //         return s; // 3
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "s").WithArguments("s").WithLocation(23, 16),
+                // (29,16): error CS8352: Cannot use variable 's' in this context because it may expose referenced variables outside of their declaration scope
+                //         return s; // 4
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "s").WithArguments("s").WithLocation(29, 16));
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71773")]
+        public void UserDefinedLogicalOperator_RefStruct_Scoped_Left()
+        {
+            var source = """
+                ref struct R
+                {
+                    private ref readonly int _i;
+                    public R(in int i) { _i = ref i; }
+                    public static bool operator true(R r) => true;
+                    public static bool operator false(R r) => false;
+                    public static R operator |(scoped R x, R y) => default;
+                }
+                class Program
+                {
+                    static R F()
+                    {
+                        return new R(1) || new R(2);
+                    }
+                }
+                """;
+            CreateCompilation(source, targetFramework: TargetFramework.Net70).VerifyDiagnostics(
+                // (13,16): error CS8347: Cannot use a result of 'R.operator |(scoped R, R)' in this context because it may expose variables referenced by parameter 'y' outside of their declaration scope
+                //         return new R(1) || new R(2);
+                Diagnostic(ErrorCode.ERR_EscapeCall, "new R(1) || new R(2)").WithArguments("R.operator |(scoped R, R)", "y").WithLocation(13, 16),
+                // (13,28): error CS8347: Cannot use a result of 'R.R(in int)' in this context because it may expose variables referenced by parameter 'i' outside of their declaration scope
+                //         return new R(1) || new R(2);
+                Diagnostic(ErrorCode.ERR_EscapeCall, "new R(2)").WithArguments("R.R(in int)", "i").WithLocation(13, 28),
+                // (13,34): error CS8156: An expression cannot be used in this context because it may not be passed or returned by reference
+                //         return new R(1) || new R(2);
+                Diagnostic(ErrorCode.ERR_RefReturnLvalueExpected, "2").WithLocation(13, 34));
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Please add a corresponding test for a binary operator as well since that hits a separate code path from binary logical operators. (Sorry, I overlooked that case when I suggested the tests above.)
```csharp
ref struct R
{
    private ref readonly int _i;
    public R(in int i) { _i = ref i; }
    public static R operator +(scoped R x, R y) => default;
}
class Program
{
    static R F()
    {
        return new R(1) + new R(2);
    }
}
```